Tests/Integration/IndexQueue/FrontendHelper/PageIndexerTest.php

```diff
@@ -194,10 +200,165 @@ public function canIndexPageIntoSolrWithAdditionalFieldsFromRootLine()
+            self::assertEquals(
+                $expectedContent,
+                $solrContent['response']['docs'][$index]['content'] ?? '',
+                'Wrong content in document ' . ($index + 1)
+            );
+        }
+    }
+
+    /**
+     * Data provider for canIndexPageWithAccessProtectedContentIntoSolr
+     */
+    public function canIndexPageWithAccessProtectedContentIntoSolrDataProvider(): Traversable
+    {
+        yield 'protected page' => [
+            'can_index_access_protected_page',
+            1,
+            [
+                '2:1/c:0',
+            ],
+            [
+                'public content of protected page',
+            ],
+        ];
+
+        yield 'protected page with protected content' => [
+            'can_index_access_protected_page_with_protected_contents',
+            2,
+            [
+                '2:1/c:0',
+                '2:1/c:2',
+            ],
+            [
+                'public content of protected page',
+                'public content of protected pageprotected content of protected page',
+            ],
+        ];
+
+        yield 'public page with protected content and global content' => [
+            'can_index_page_with_protected_content',
+            2,
+            [
+                'c:0',
+                'c:1',
+            ],
+            [
+                'public ce',
+                'protected cepublic ce',
+            ],
+        ];
+
+        yield 'public page with protected and hide at login content' => [
+            'can_index_page_with_protected_and_hideatlogin_content',
+            2,
+            [
+                'c:0',
+                'c:1',
+            ],
+            [
+                'hide at login content',
+                'protected ce',
+            ],
+        ];
+    }
+
+    /**
+     * Sends a page indexer request
+     *
+     * In test environment we have to use an InternalRequest, this method
+     * is intended to replace PageIndexerRequest->send()
+     */
+    protected function sendPageIndexerRequest(string $url, PageIndexerRequest $request): PageIndexerResponse
+    {
+        $internalRequest = new InternalRequest($url);
+
+        foreach ($request->getHeaders() as $header) {
+            [$headerName, $headerValue] = GeneralUtility::trimExplode(':', $header, true, 2);
+            $internalRequest = $internalRequest->withAddedHeader($headerName, $headerValue);
+        }
+
+        $rawResponse = $this->executeFrontendSubRequest($internalRequest);
+        $rawResponse->getBody()->rewind();
+
+        $indexerResponse = GeneralUtility::makeInstance(PageIndexerResponse::class);
+        $decodedResponse = $indexerResponse->getResultsFromJson($rawResponse->getBody()->getContents());
+        $rawResponse->getBody()->rewind();
+        if ($decodedResponse === null) {
+            throw new RuntimeException('Failed to execute Page Indexer Request during integration test');
+        }
```

Comment:
Please use `self::assertNotNull() here`
```suggestion
        self::assertNotNull($decodedResponse, 'Failed to execute Page Indexer Request during integration test');
```